src/token_request_handler.ts

```diff
@@ -56,14 +88,18 @@ export class BaseTokenRequestHandler implements TokenRequestHandler {
       data: this.utils.stringify(request.toStringMap())
     });
 
-    return tokenResponse.then(response => {
-      // check for error response
-      if (this.isTokenResponse(response)) {
-        return TokenResponse.fromJson(response);
-      } else {
-        return Promise.reject<TokenResponse>(
-            new AppAuthError(response.error, TokenError.fromJson(response)));
-      }
-    });
+    return tokenResponse
+        .then(response => {
+          if (this.isTokenResponse(response)) {
+            return TokenResponse.fromJson(response);
+          } else {
+            return Promise.reject<TokenResponse>(
+                new AppAuthError(response.error, TokenError.fromJson(response)));
+          }
+        })
+        .catch(response => {
```

Comment:
Why are you re-catching and re-throwing it as another AppAuth error. Seems unnecessary.